ucp/_libs/ucx_api.pyx

```diff
@@ -386,6 +387,64 @@ cdef class UCXWorker(UCXObject):
         assert_ucs_status(status)
         return UCXEndpoint(self, <uintptr_t>ucp_ep)
 
+    def fence(self):
+        cdef ucs_status_t status
+        status = ucp_worker_fence(self._handle)
+        assert_ucs_status(status)
+        return status
+
+    def flush(self, cb_func, cb_args, cb_kwargs=dict()):
+        cdef ucs_status_ptr_t req
+        cdef ucp_send_callback_t _send_cb = <ucp_send_callback_t>_send_callback
+
+        cdef ucs_status_ptr_t status = ucp_worker_flush_nb(self._handle, 0, _send_cb)
+        return _handle_status(
+            status, 0, cb_func, cb_args, cb_kwargs, 'flush', self._inflight_msgs
+        )
+
+    def get_address(self):
+        return UCXAddress(self)
+
+
+def _ucx_address_finalizer(uintptr_t handle_as_int, UCXWorker worker):
+    cdef ucp_address_t *address = <ucp_address_t *>handle_as_int
+    cdef ucp_worker_h ucp_worker = <ucp_worker_h><uintptr_t>worker.handle
+    ucp_worker_release_address(ucp_worker, address)
+
+
+cdef class UCXAddress(UCXObject):
+    """Python representation of ucp_address_t"""
+    cdef ucp_address_t *_handle
+    cdef size_t _length
+    cdef worker
+    __array_interface__ = dict()
+
+    def __init__(self, UCXWorker worker):
+        cdef ucs_status_t status
+        cdef ucp_worker_h ucp_worker = <ucp_worker_h><uintptr_t>worker.handle
+
+        status = ucp_worker_get_address(ucp_worker, &self._handle, &self._length)
+        assert_ucs_status(status)
+        self.worker = worker
+        self.__array_interface__["data"] = (<uintptr_t>self._handle, True)
+        self.__array_interface__["typestr"] = "|V1"
+        self.__array_interface__["shape"] = [self._length]
+        self.__array_interface__["version"] = 3
```

Comment:
This can be dropped as the Python Buffer Protocol implementation below handles this.

Reply:
I always meant to do this as a buffer protocol, but the first time I tried that didn't work out so well so I went with the numpy array interface to 'just make it work'. I think I work with your suggestion below.